### PR TITLE
docs: improve description of rev requirements for post/put document

### PR DESCRIPTION
**Pull request by mojito317**
## PR summary

The goal of this PR is to improve the description of `_rev`/`rev` requirements for post/put document. I also fixed the ornamentation comment lines to 82 char length because I caught those not shared the same length. I also tried to explain where `_id`/`docId` is necessary similarly to `_rev`/`rev`.

I decided to use `/*...code is here...*/` for code comments and ornamentation lines before and after the explanation plus the commented out codes because with this way maybe the eyes can catch what belongs to what. Something like this:

```
// ========================================================================
// Note: the next code bla bla bla..:
/*
 code is here
*/
// ========================================================================
```

❗ REVIEWERS ❗ 
Please read the `Other information` section before review

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [x] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?
Lack of explanation around the usage of id and rev fields in CUD operation examples.

## What is the new behavior?
Explanation in example code how to use `putDocument` along with `postDocument` and:
* `_id` is a MUST at CREATE for `putDocument`
* `_id` is OPTIONAL at CREATE for `postDocument`
* `_id` is generated at CREATE when it is not provided `postDocument`
* `_id` and `_rev` is a MUST at UPDATE
*  `id` parameter is a MUST at UPDATE for `putDocument`
* `rev` parameter can be replaced by `_rev` in the document object both for `postDocument` and `putDocument`
* `id` and `rev` parameter is a MUST at DELETE 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

❗ REVIEWERS ❗ 
I would like to ask you to review only `*.js` files in the first round. This way we can discuss wording in one place and I will fix them in both `.js` and `.ts` files. Thanks in advance!

### CODE REVIEW

File: test/examples/src/js/UpdateDoc.js [hunk 1]
```diff
@@ -36,35 +36,60 @@ const updateDoc = async () => {
...
+    // ==========================================================================
 
     // Add Bob Smith's address to the document
     document.address = '19 Front Street, Darlington, DL5 1TY';
 
     // Remove the joined property from document object
     delete document['joined'];
 
-    // Keeping track with the revision number of the document object:
+    // Keeping track with the latest revision number of the document object
```
Comment (emlaver) [inline test/examples/src/js/UpdateDoc.js:57]:
```suggestion
    // Keeping track of the latest revision number of the document object
```

File: test/examples/src/js/UpdateDoc.js [hunk 2]
```diff
@@ -36,35 +36,60 @@ const updateDoc = async () => {
...
+    // is necessary for further UPDATE/DELETE operations:
+    document._rev = (
+      await client.postDocument({
+        db: exampleDbName,
+        document, // _id and _rev MUST be inside the document object
```
Comment (emlaver) [inline test/examples/src/js/UpdateDoc.js:62]: I think we can remove the comma
```suggestion
        document // _id and _rev MUST be inside the document object
```

File: test/examples/src/js/UpdateDoc.js [hunk 3]
```diff
@@ -36,35 +36,60 @@ const updateDoc = async () => {
...
 
-    // Note: for request byte stream use:
-    // document._rev = (
-    //     await client.postDocument({
-    //       db: exampleDbName,
-    //       document: documentAsByteStream,
-    //     })
-    // ).result.rev;
+    // ==========================================================================
+    // Note 2: updating the document be also done with the "putDocument" function.
```
Comment (emlaver) [inline test/examples/src/js/UpdateDoc.js:79]:
```suggestion
    // Note 2: updating the document can also be done with the "putDocument" function.
```

File: test/examples/src/ts/CreateDbAndDoc.ts [hunk 4]
```diff
@@ -45,26 +45,39 @@ const createDb = client
...
-// Save the document in the database
+// Save the document in the database with "postDocument" function
 createDb.then(() => {
   client
     .postDocument({
       db: exampleDbName,
       document: exampleDocument,
     })
+    // ==========================================================================
+    // Note: saving the document could have been also done with the "putDocument"
```
Comment (emlaver) [inline test/examples/src/ts/CreateDbAndDoc.ts:68]:
```suggestion
    // Note: saving the document can also be done with the "putDocument"
```

File: test/examples/src/ts/CreateDbAndDoc.ts [hunk 5]
```diff
@@ -45,26 +45,39 @@ const createDb = client
...
+    .putDocument({
+      db: exampleDbName,
+      docId: exampleDocId,
+      document: exampleDocument,
+    })
+    */
+    // ==========================================================================
     .then((createDocumentResponse) => {
-      // Keep track with the revision number of the document object
+      // Keeping track with the revision number of the document object
```
Comment (emlaver) [inline test/examples/src/ts/CreateDbAndDoc.ts:79]:
```suggestion
      // Keeping track of the revision number of the document object
```

File: test/examples/src/ts/UpdateDoc.ts [hunk 6]
```diff
@@ -59,12 +63,29 @@ client
...
-      // Note: for request byte stream use:
+      // ========================================================================
+      // Note 1: for request byte stream use:
       // .postDocument(
       //   {db: exampleDbName, document: documentAsByteStream}
       // )
+      // ========================================================================
+
+      // ========================================================================
+      // Note 2: updating the document be also done with the "putDocument" function.
```
Comment (emlaver) [inline test/examples/src/ts/UpdateDoc.ts:74]:
```suggestion
      // Note 2: updating the document can also be done with the "putDocument" function.
```

File: test/examples/src/ts/UpdateDoc.ts [hunk 7]
```diff
@@ -59,12 +63,29 @@ client
...
+        db: exampleDbName,
+        docId: document._id, // docId is a required parameter
+        rev: document._rev,
+        document, // _rev in the document object CAN replace above `rev` parameter
+      })
+      */
+      // ========================================================================
       .then((res) => {
-        // Keeping track with the revision number of the document object:
+        // Keeping track with the latest revision number of the document object
```
Comment (emlaver) [inline test/examples/src/ts/UpdateDoc.ts:87]:
```suggestion
        // Keeping track of the latest revision number of the document object
```

**Review by emlaver — COMMENTED** [review]
Looks good!

File: test/examples/src/js/UpdateDoc.js (hunk 2 above)
Comment (mojito317) [reply test/examples/src/js/UpdateDoc.js:62]: This is added by prettier on save.

File: test/examples/src/js/CreateDbAndDoc.js [hunk 8]
```diff
@@ -49,19 +49,32 @@ const createDbAndDoc = async () => {
...
-  // Save the document in the database
+  // Save the document in the database with "postDocument" function
   const createDocumentResponse = await client.postDocument({
     db: exampleDbName,
     document: exampleDocument,
   });
 
-  // Keep track with the revision number of the document object
+  // ==========================================================================
+  // Note: saving the document could have been also done with the "putDocument"
```
[the remark this answers is not in the document]
Comment (mojito317) [reply test/examples/src/js/CreateDbAndDoc.js]: Done in f526dc7.

File: test/examples/src/js/UpdateDoc.js (hunk 1 above)
Comment (mojito317) [reply test/examples/src/js/UpdateDoc.js]: Done in f526dc7.

File: test/examples/src/js/UpdateDoc.js (hunk 3 above)
Comment (mojito317) [reply test/examples/src/js/UpdateDoc.js]: Done in f526dc7.

File: test/examples/src/ts/CreateDbAndDoc.ts (hunk 4 above)
Comment (mojito317) [reply test/examples/src/ts/CreateDbAndDoc.ts]: Done in f526dc7.

File: test/examples/src/ts/CreateDbAndDoc.ts (hunk 5 above)
Comment (mojito317) [reply test/examples/src/ts/CreateDbAndDoc.ts]: Done in f526dc7.

File: test/examples/src/ts/UpdateDoc.ts (hunk 6 above)
Comment (mojito317) [reply test/examples/src/ts/UpdateDoc.ts]: Done in f526dc7.

File: test/examples/src/ts/UpdateDoc.ts (hunk 7 above)
Comment (mojito317) [reply test/examples/src/ts/UpdateDoc.ts]: Done in f526dc7.

**Review by emlaver — APPROVED** [review]

**Review by tomblench — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews @emlaver and @tomblench. I'm about to merge this!

**Pull request merged**